### DISCUSSION TITLE
Question Paper Preview Before Finalisation

### DISCUSSION THREAD

**Discussion by tarang27** [Sunbird-coKreat/Community#13] [Feature Requests]
Hi All,

Sharing a brief note on the new feature that we have planned as part of Prashnavali solution on VDN for the 5.0 release.

In the current flow, each question undergoes the following steps:- 

1. Question creator creates a question
2. Contribution reviewer accepts/rejects the question
3. Sourcing reviewer/Question paper creator approves/rejects the question

In the third stage, where a question is getting selected to be added in a question paper a preview of the list of questions is not available at one place. This creates an issue for the question paper creator as he/she is unable to gauge two things:-

- Will I be able to select at-least the required number of questions to be added in the question paper ?
- Will I be able to get the best questions out of the reviewed questions added in the question paper ?

The reason why these questions remain unanswered for the question paper creator are two fold:-

- The question paper creator is unable to get a single window view of the lot of questions that have been reviewed on the contribution portal, so he/she cannot make the judgement whether making an effort to review each question individually would result in getting the required number of questions.

- To select (Approve/Reject) each individual question, the question paper creator has to click on the question card from the list view of questions, go through all the details of that question (body of the question, correct answer & relevant tags) & decide whether that particular question can be selected to the question paper, without knowing if better questions are available or if the required number of questions have already been selected.

**Functional solution:**

Enable a preview for the question paper creator on the sourcing portal, of all the questions accepted from the review that has happened on the contribution portal.

A detailed PRD has also been created for the same:- https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3196878857/Question+Paper+Preview+Before+Finalisation

Would request you to go through it and share your comments
@surendrasinghs @pallakartheekreddy

**Comment by Prerna789** [comment]
[@vrayulu](https://github.com/vrayulu)
[@maheshkumargangula](https://github.com/maheshkumargangula)
